### ISSUE TITLE
azurerm_dns_cname_record is setting records as empty

### ISSUE THREAD

**Issue by timhaak** [hashicorp/terraform#7101]
### Terraform Version
Terraform v0.6.16

### Affected Resource(s)
- azurerm_dns_cname_record
Everything else is working ie a records etc.

### Terraform Configuration Files
```hcl

resource "azurerm_dns_cname_record" "testing" {
  name = "testing"
  zone_name = "${azurerm_dns_zone.tulun.name}"
  resource_group_name = "${azurerm_resource_group.production-we.name}"
  ttl = "3600"
  records = ["str.tulun.tv"]
  lifecycle = {
    prevent_destroy = true
  }
}

```

### Apply output

```
azurerm_dns_cname_record.testing: Creating...
  name:                "" => "testing"
  records.#:           "" => "1"
  records.488963031:   "" => "str.tulun.tv"
  resource_group_name: "" => "tulres01"
  ttl:                 "" => "3600"
  zone_name:           "" => "tulun.tv"
azurerm_dns_cname_record.testing: Creation complete
```
### Expected Behavior
That the record/alias is set.

### Actual Behavior
Record created but set alias set to an empty string

### Steps to Reproduce
terraform apply

### Important Factoids
Nothing strange that I'm aware of

### Screen shot from azure of output
https://www.dropbox.com/s/lz3ddjvg3az2abk/Screenshot%202016-06-10%2011.24.41.png?dl=0

**Comment by stack72** [comment]
Hi @timhaak 

I am currently looking into this to see what i can find out. 

Paul

**Comment by timhaak** [comment]
Cool thanks. 

Let me know if there is something I can test.

**Comment by stack72** [comment]
ok, have recreated and found the same thing - this means it's debug time :)

**Comment by stack72** [comment]
@timhaak 

ok, we have a working solution, the error was because the Azure API changed to only allow 1 record and we were passing in multiple

So I need to work out how to migrate people's state from the old way to the new way, I will try and get this in place for 0.7

Thanks

Paul

**Comment by timhaak** [comment]
Ok cool 

Not urgent on my side logged it more to help for the future.

Tim Haak
+27 83 778 7100
tim@haak.co <mailto:tim@haak.co>

**Comment by stack72** [comment]
The PR that fixes this is #7113 

there is going to be a change to the file records[] -> record

**Comment by stack72** [comment]
Fixed in #7113
